stackit/internal/services/argus/instance/resource.go

```diff
@@ -469,6 +727,13 @@ func (r *instanceResource) Read(ctx context.Context, req resource.ReadRequest, r
 		return
 	}
 
+	// Map response body to schema
+	err = mapAlertConfigField(ctx, alertConfigResp, &model)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating instance", fmt.Sprintf("Processing API response for the alert config: %v", err))
+		return
+	}
```

Comment:
Since the field is only Optional (and not Computed), do we need to map the alert configs to the state in the Read? Isn't this lead to the plan always showing changes?

Reply:
This still needs to be mapped, there are some fields that are set by default by the API and are not required, for instance the `globalOptions` or some fields inside the `emailConfigs` etc, so we would need to update the state accordingly

The part about leading to the plan always showing changes I don't completely understand. What does happen is that if only one of the optional fields is set, the plan will show `(state known after apply)` in the others, but the plan after an apply is empty since the fields already match what was read, right?
Even in the case of the mock config, since we just set it to `Null` it always matches as well

Reply:
Right, because we map to null objects if the value is the mocked config